### ISSUE TITLE
(bug) The player keeps moving when the TalkDialog is called

### ISSUE THREAD

**Issue by TaylorHo** [RafaelBarbosatec/bonfire#271]
## Introducing the bug
When approaching some items on the map, I call ```TalkDialog.show``` to show an interactive dialog with the item.
When used on non-mobile platforms (such as web and desktops), when displaying dialog, ```KeyboardDirectional``` crashes and keeps the player moving nonstop.

## Way to prevent (which still needs improvement)
To prevent this, you can use ```gameRef.player?.stopMoveAlongThePath();```, but as soon as the dialog closes and the player starts walking again, it locks the direction again, moving indefinitely.

## Additional Info
It is worth mentioning that the error only occurs with ```KeyboardDirectional``` on the joystick (used on the web, Windows, Linux, and macOS).

---

## Example:

```dart
    // Stop the player (this line is needed bc without it, the player movement locks on the last used direction)
    gameRef.player?.stopMoveAlongThePath();

    // Show the Dialog
    TalkDialog.show(
      context,
      [
        Say(
          text: [
            TextSpan(
              text: 'example text',
            ),
          ],
        ),
      ],
    );

    // After closing the dialog, no matter which direction you go, the player keeps going in the last direction (from before the dialog) + the direction chosen now
    // Example: Last used direction is [up], and the direction chosen after the dialog end is [right], then, the player will go to up + right direction (diagonal).
    // The direction only returns to normal after going in the same direction the player had gone before the crash (going in the same direction as before the dialogue results in an "unlock" of the direction)
```

## Final considerations
Here we have two changes to be made/fixed:
- Automatically stop player when some dialog is shown;
- Fix the lock of the player's direction, after the end of the dialog.

Thanks a lot!

**Comment by RafaelBarbosatec** [comment]
Hi @TaylorHo !
Thanks for this report. 
I'm still trying to reproduce. I haven't been able to catch it yet.

**Comment by TaylorHo** [comment]
Ohh, Ok!
I'll try to send here a minimal reproducible project/code, demonstrating the bug.

**Comment by TaylorHo** [comment]
### Reproducible Code

Download the compressed project attached below, run the ```flutter pub get```, and then, run the **web version of the project** (usually ```flutter run -d chrome```).
Just remembering that the error only occurs on desktop platforms, such web.


I've added a better explanation at the ```lib/fire.dart``` file (where the TalkDialog is shown).

[example.zip](https://github.com/RafaelBarbosatec/bonfire/files/9129052/example.zip)

**Comment by rafaelbarbosadrip** [comment]
Fixed in version 2.6.4. Pls use `gameRef.player?.idle();` to stop player.
